### PR TITLE
Splunk otel collector package

### PR DESCRIPTION
**Pull request by utieyin**
splunk-otel-collector: new package

#### For new package PRs only

- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates

### CODE REVIEW

File: splunk-otel-collector.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    - license: Apache-2.0
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/signalfx/splunk-otel-collector
+      tag: v${{package.version}}
+      expected-commit: afd65d06be2164c95fb50a9d6ab2f42af9e86383
+
+  - uses: go/build
```
[the remark this answers is not in the document]
Comment (developer-guy) [reply splunk-otel-collector.yaml:16]: https://github.com/signalfx/splunk-otel-collector/blob/main/Makefile#L149

File: splunk-otel-collector.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    with:
+      modroot: cmd/otelcol
+      packages: .
+      ldflags: -X github.com/signalfx/splunk-otel-collector/internal/version.Version=v${{package.version}}
```
Comment (developer-guy) [inline splunk-otel-collector.yaml:20]: it looks like there are two ldflags should be defined: https://github.com/signalfx/splunk-otel-collector/blob/main/Makefile#L30-L32
this one `-X go.opentelemetry.io/collector/internal/version.Version=${{package.version}}` needs to be added as well.

File: splunk-otel-collector.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (utieyin) [reply splunk-otel-collector.yaml:16]: @developer-guy I agree, just couldn't find where it was been used in the actual image.

File: splunk-otel-collector.yaml (hunk 2 above)
Comment (utieyin) [reply splunk-otel-collector.yaml]: Both ldflags added, thanks for catching this

File: splunk-otel-collector.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (utieyin) [reply splunk-otel-collector.yaml:23]: This is now been moved to a subpackage, thanks

**Review by developer-guy — APPROVED** [review]

**Pull request merged**